cmd/flux/tree_kustomization.go

```diff
@@ -284,5 +289,35 @@ func getHelmReleaseInventory(ctx context.Context, objectKey client.ObjectKey, ku
 		}
 	}
 
-	return object.UnstructuredSetToObjMetadataSet(objects), nil
+	result := object.UnstructuredSetToObjMetadataSet(objects)
+
+	// search for CRDs managed by the HelmRelease if installing or upgrading CRDs is enabled in spec
+	if (hr.Spec.Install != nil && hr.Spec.Install.CRDs != "") ||
+		(hr.Spec.Upgrade != nil && hr.Spec.Upgrade.CRDs != "") {
+		selector := client.MatchingLabels{
+			fmt.Sprintf("%s/name", helmv2.GroupVersion.Group):      hr.GetName(),
+			fmt.Sprintf("%s/namespace", helmv2.GroupVersion.Group): hr.GetNamespace(),
+		}
+		var list apiextensionsv1.CustomResourceDefinitionList
```

Comment:
We need to investigate why using `metav1.PartialObjectMetadataList` instead of `apiextensionsv1.CustomResourceDefinitionList` returns an empty list. For now I will go with this version.